src/main/java/core/basesyntax/dao/ProductDao.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+public interface ProductDao {
```

Comment:
You are accesing Storage data, but not product's.  I think it is better to rename this class.